src/main/java/robot/subsystems/drivetrain/pure_pursuit/PurePursue.java

```diff
@@ -30,50 +44,74 @@ public PurePursue(Path path) {
     protected void initialize() {
         lastLeftEncoder = drive.getLeftDistance();
         lastRightEncoder = drive.getRightDistance();
-        initAngle = Robot.navx.getAngle();
+        //initAngle = drive.getAngle() + (direction == -1 ? 180 : 0);
+        currentLookahead = path.getWaypoint(0);
+        lastLeftSpeed = direction * drive.getLeftSpeed();
+        lastRightSpeed = direction * drive.getRightSpeed();
     }
 
     // Called repeatedly when this Command is scheduled to run
     protected void execute() {
         updatePoint();
-
+        updateLookaheadInPath(path);
+        /*
+        Main methods:
+            updatePoint()
+            updateLookaheadInPath()
+            closest Point
+         */
 
     }
 
     // Make this return true when this Command no longer needs to run execute()
     protected boolean isFinished() {
-        return false;
+        return (closestPoint(path).getX() == path.getWaypoint(path.length() - 1).getX() && closestPoint(path).getY() == path.getWaypoint(path.length() - 1).getY());
```

Comment:
This is not enough. Imagine the robot passing through the endpoint at a speed of 100 km/h. The command will end, but we cannot say the robot has finished the pursuit. You should add a condition checking whether the robot has actually stopped at the endpoint.
Also, [IIAC](https://www.urbandictionary.com/define.php?term=IIAC), it seems that you're checking whether the robot has reached exactly the endpoint. This would probably never occur, so you should add a small deadband. The same, for the speeds,
```suggestion
        return (closestPoint(path).getX() == path.getWaypoint(path.length() - 1).getX() && closestPoint(path).getY() == path.getWaypoint(path.length() - 1).getY() && Robot.drivetrain.getLeftSpeed() == 0 && Robot.drivetrain.getRightSpeed() == 0);
```

Reply:
Also i love the fact that you used an acronym to save time but then linked a link to the acronyms meaning 😄

Reply:
Well, I hope it would be the last time I'd need to link the meaning 😁.